### PR TITLE
Fixes abbreviated genitive

### PR DESCRIPTION
**Pull request by lillo42**
Fixes bugs: https://github.com/dotnet/corefx/issues/2804
PR on CoreFX: https://github.com/dotnet/corefx/pull/33695

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -37,6 +37,9 @@ msbuild.log
 *.sln.ide
 *.ide/
 
+# Rider
```
Comment (tarekgh) [inline .gitignore]: I don't think this change is needed.

**Comment by tarekgh** [comment]
test Linux-musl x64 Debug Build

File: src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormat.cs [hunk 2]
```diff
@@ -630,13 +630,13 @@ private static bool IsUseGenitiveForm(ReadOnlySpan<char> format, int index, int
                             }
                             else
                             {
-                                if ((dtfi.FormatFlags & DateTimeFormatFlags.UseGenitiveMonth) != 0 && tokenLen >= 4)
+                                if ((dtfi.FormatFlags & DateTimeFormatFlags.UseGenitiveMonth) != 0 && tokenLen >= 3)
```
[the remark this answers is not in the document]
Comment (tarekgh) [reply src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormat.cs]: good catch, the check was corret when we were testing 4 but now we don't really need this check.

---
In reply to: [237103540](https://github.com/dotnet/coreclr/pull/21230#discussion_r237103540) [](ancestors = 237103540)

**Comment by tarekgh** [comment]
@lillo42 could you please update the code according to @stephentoub comment?

**Comment by lillo42** [comment]
@tarekgh yes, of course, I will do as soon as possible.

**Comment by tarekgh** [comment]
test OSX10.12 x64 Checked Innerloop Build and Test

**Comment by tarekgh** [comment]
test OSX10.12 x64 Checked Innerloop Build and Test

**Pull request merged**